Improve Satellite automation while discovering it

ericzolf opened the pull request:
It's a bit of a catch all but I fixed/improved all this as I went through the process of creating my first Satellite using this repo, so here you are:

- ignore working and private files 'my own' in Git
- add small HOWTO on how to create a Satellite
- rename group from 'satellite' to 'satellites' (plural for groups)
- add become: true and tags sat_* to playbooks/sat_install.yml so that it's easier to select parts of it
- replace Ansible 2.8 repo through Ansible 2 so that we don't depend on latest version
- remove duplicate variables in roles/install/defaults/main.yml
- add variable satellite_force_install to the install role to be able to restart a failed installation
- multiple typos
- disabled the tuning file which is rejected by Satellite 6.8 because not valid for Postgres 9.5+
- use template instead of file+blockinfile to create the disconnected.repo
- moved the enabled repos to vars/main.yml in rhsm role so that it is easier to maintain/oversee
- extended run_sat.sh to accept SAT_ENV_FILE as environment file and be able to use ansible-playbook parameters
- small variable names corrections in sat_env.sh and more explanations

At the time of writing, I'm still waiting for the repos to finish synchronizing but it looks all good now.

ericzolf committed:
Improve Satellite automation while discovering it

It's a bit of a catch all but I fixed/improved all this as I went through the process of creating my first Satellite using this repo, so here you are:

- ignore working and private files 'my own' in Git
- add small HOWTO on how to create a Satellite
- rename group from 'satellite' to 'satellites' (plural for groups)
- add become: true and tags sat_* to playbooks/sat_install.yml so that it's easier to select parts of it
- replace Ansible 2.8 repo through Ansible 2 so that we don't depend on latest version
- remove duplicate variables in roles/install/defaults/main.yml
- add variable satellite_force_install to the install role to be able to restart a failed installation
- multiple typos
- disabled the tuning file which is rejected by Satellite 6.8 because not valid for Postgres 9.5+
- use template instead of file+blockinfile to create the disconnected.repo
- moved the enabled repos to vars/main.yml in rhsm role so that it is easier to maintain/oversee
- extended run_sat.sh to accept SAT_ENV_FILE as environment file and be able to use ansible-playbook parameters
- small variable names corrections in sat_env.sh and more explanations

ericzolf committed:
Few more fixes to get content and config working

- align some more names of repos for Ansible and Satellite
- add an environment variable SATELLITE_SUBS_FOR_AK to be able to configure externally the subscriptions added by default

ericzolf committed:
Clarify that steps create a 'standard' Satellite

ericzolf commented:
I pushed two additional commits to address few more points:

- align some more names of repos for Ansible and Satellite
- add an environment variable SATELLITE_SUBS_FOR_AK to be able to configure externally the subscriptions added by default
- clarify that the howto creates a "standard" Satellite

evgeni commented on roles/config/defaults/main.yml:
```diff
@@ -36,12 +36,12 @@ satellite_content_views:
     repositories:
       # yamllint disable-line rule:line-length
-      - name: "Red Hat Ansible Engine 2.8 RPMs for Red Hat Enterprise Linux 7 Server x86_64"
+      - name: "Red Hat Ansible Engine 2 RPMs for Red Hat Enterprise Linux 7 Server x86_64"
```
technically, every satellite version has a specific supported ansible version (6.7: 2.8, 6.8: 2.9, etc), so not sure this is a good idea -- not that the old code was better ;)

ericzolf replied:
I could add a dictionary in the vars, mapping the Satellite version to the Ansible version. Still a bit ugly but less... What do you think @evgeni

evgeni replied:
We had a similar discussion with @ehelms in a different repo and ended up on "make the user enter two things for now", mostly for simplicity's sake.

Not against a dict lookup, just also not thinking it's explicitly required :)

ehelms replied:
Yea, I'm torn. Here, the lifecycle of a Satellite version is longer, and the support is well defined so I would be OK with a dict to specify this and simplify it for a user. Since Satellite version X supports Ansible version Y, we shouldn't allow users to make a choice and get it wrong is where I am leaning.

ericzolf replied:
The last added commit defines a default based on the mapping outlined by Evgeni but can be overwritten if the user of the automation "knows better". Ansible "2" is the default if the mapping isn't defined, e.g. for 6.9, which should be fine in most cases.

ericzolf committed:
Introduce variable for Ansible version dependent from Satellite version

The variable satellite_ansiver is dependent on the Satellite version (by default)
by using the satellite_ansible_map variable in vars.

ericzolf committed:
Merge branch 'master' into ericzolf-various-fixes
